rfc-113-expanding-draft-access-for-unauthenticated-users.md

```diff
@@ -0,0 +1,212 @@
+# Expanding draft access for unauthenticated users to allow multi-page fact checks
+
+## Summary
+
+To view content on the [GOV.UK draft stack][govuk-draft] users have to either
+be authenticated with [GOV.UK signon][signon] or have a token that grants
+them access to that piece of content. Sensitive content is flagged as
+access limited, this limits drafts to authenticated users that meet a criteria
+(user id or organisation id) or users with the token.
+
+The reason for granting unauthenticated access is to make it easier for
+non-GOV.UK editors to fact check content, avoiding the overhead of needing a
+GOV.UK signon account creation and access to all of draft GOV.UK. Evidence
+suggests that without a simple process to fact check, editors will instead
+turn to a poorer fact check experience (for example, screenshots or pasting
+content into an email).
+
+This RFC makes a proposal to expand the unauthenticated access from a single
+piece of content to scenarios where content can span multiple pages from
+differing publishing applications. This is needed to support the fact checking
+of a step-by-step and involves using a session cookie to store an auth-bypass
+token so multiple pages and their assets can be accessed. Access to content
+will be granted by using Publishing API [link expansion][] to share an
```

Comment:
It ends up being both dependency resolution and link expansion. Dependency resolution determines that those pages need to be updated and link expansion is used to work out the data in those links.

The `auth_bypass_id` would be in the links of the content like so:

```
{
  "content_id": "12345-1234-1234-1231123",
  "auth_bypass_ids": ["dfghjklkjhgf4567897654"],
  "access_limit": { ... },
  "expanded_links": {
    "pages_part_of_step_nav": [
      {
         "api_path": "/api/content/apply-first-provisional-driving-licence",
         ...
         "auth_bypass_ids": ["1234675sdfgsdfghsdfgh765yrtfg"],
      }
  }
```

Logic in content store would first check the content items own `auth_bypass_ids` as they can bypass access limits. Then it would check the presence of `auth_bypass_ids` in the links.